draft-ietf-tls-esni.md

```diff
@@ -535,6 +532,26 @@ application using (D)TLS or externally configured on both sides,
 implementations MUST compute the field as specified in
 {{encrypted-client-hello}}.
 
+### IsClientHelloInner Extension {#is-inner}
+
+If, in a ClientHello, the "encrypted_client_hello" extension is not present and
+an "is_client_hello_inner" extension is present, the ClientHello is a
+ClientHelloInner. This extension MUST only be sent in the ClientHello message.
```

Comment:
I don't feel strongly about the name, though this extension seems to me more about "swizzling" ServerHello.random (by setting the confirmation signal), and less about the fact that the CH is of type ClientHelloInner. Unless there's other things that backend servers need to do in response to a CHInner (beyond putting in the acceptance signal), I'd refocus this extension so it focuses just on that piece.

Reply:
It took a LOT of reading and re-reading the "swizzle" comment to understand what it was referring to, and I imagine it would be the same for people who are new to this. To me, it's clearer to say "this is just a sign that this ClientHello is an inner ClientHello," and then leave it to the backend server to figure out what to do about it.

Reply:
I like the name, for the record.